crates/typst/src/layout/place.rs

```diff
@@ -136,3 +144,31 @@ impl Behave for Packed<PlaceElem> {
         Behaviour::Ignorant
     }
 }
+
+/// Asks the layout algorithm to place pending floating elements before
+/// continuing with the content.
+///
+/// This is useful for preventing floating figures from spilling
+/// into the next section.
+///
+/// ```example
+/// #set page(height: 165pt, width: 150pt)
+///
+/// Some introductory text: #lorem(15)
+///
+/// #figure(placement: auto, caption: [A self-describing figure], rect(width: 100%, height: 64pt, [I float with a caption!]))
+///
+/// #place.flush()
+///
+/// Some conclusive text that must occur after the figure.
```

Comment:
The content is better, the number of columns is still very high. There will likely be a bad linebreak when rendered. I suggest something like this:
```suggestion
/// #set page(height: 165pt, width: 150pt)
///
/// Some introductory text: #lorem(15)
///
/// #figure(
///   rect(
///     width: 100%, 
///     height: 64pt, 
///     [I float with a caption!],
///   ),
///   placement: auto, 
///   caption: [A self-describing figure], 
/// )
///
/// #place.flush()
///
/// Some conclusive text that must occur 
/// after the figure.
```